extensions/src/main/java/org/pentaho/platform/plugin/services/pluginmgr/PentahoSystemPluginManager.java

```diff
@@ -365,12 +369,24 @@ private void registerPlugin( final IPlatformPlugin plugin ) throws PlatformPlugi
 
     ClassLoader loader = PentahoSystem.get( ClassLoader.class, null,
       Collections.singletonMap( PLUGIN_ID, plugin.getId() ) );
-    GenericApplicationContext beanFactory = PentahoSystem
-      .get( GenericApplicationContext.class, null, Collections.singletonMap( PLUGIN_ID, plugin.getId() ) );
+
+    Objects.requireNonNull( loader );
+
+    var baseName = plugin.getResourceBundleClassName();
```

Comment:
Better use a more specific variable name, in this scope, e.g. `resourceBundleBaseName`?